pkg/workflow/controller/handlers/workflowrun/handler.go

```diff
@@ -165,40 +91,47 @@ func (h *Handler) ObjectUpdated(old, new interface{}) {
 		if err != nil {
 			log.WithField("wfr", originWfr.Name).Warn("send notification failed", err)
 		}
-		return
+		return nil
+	}
+
+	// Add this WorkflowRun to timeout processor, so that it would be cleaned up when time expired.
+	err := h.TimeoutProcessor.AddIfNotExist(originWfr)
+	if err != nil {
+		log.WithField("wfr", originWfr.Name).Warn("add wfr to timeout processor failed: ", err)
 	}
 
-	// If the WorkflowRun has already been terminated or waiting for external events, skip it.
-	if originWfr.Status.Overall.Phase == v1alpha1.StatusSucceeded ||
-		originWfr.Status.Overall.Phase == v1alpha1.StatusFailed ||
-		originWfr.Status.Overall.Phase == v1alpha1.StatusWaiting {
-		return
+	// If the WorkflowRun is waiting for external events, skip it.
```

Comment:
Still need to handle terminated wfr?

Reply:
The terminated wfr has been handled [before here](https://github.com/caicloud/cyclone/pull/1430/files/2390f9d0a85d8260f229c1e4c46e668c6865d70b#diff-1f9eb1fba2691104d8ecb95959b80315R85).